src/library/vm/vm.h

```diff
@@ -48,7 +49,7 @@ class vm_obj_cell {
 };
 
 #define LEAN_VM_IS_PTR(obj) ((reinterpret_cast<size_t>(obj) & 1) == 0)
-#define LEAN_VM_BOX(num)    (reinterpret_cast<vm_obj_cell*>((num << 1) | 1))
+#define LEAN_VM_BOX(num)    (reinterpret_cast<vm_obj_cell*>((uintptr_t)((num << 1) | 1)))
 #define LEAN_VM_UNBOX(obj)  (reinterpret_cast<size_t>(obj) >> 1)
```

Comment:
Why is this needed?

Reply:
because num is an unsigned, and thus 32-bits in VS (and others). When doing a reinterpret_cast between types of different sizes, VS gives an error. uinptr_t is the integer type with the same size as pointers, so the problem is fixed.

Reply:
I see. We use `static_cast<size_t>(n)` for this purpose in the Lean code base.
Could you please replace it with this?

Reply:
ok, done

Reply:
BTW, I have been using `size_t` instead of `uintptr_t`, and always assumed that a pointer would fit into `size_t`, but I just realized this is not true. https://stackoverflow.com/questions/1572099/size-t-is-pointer-size-in-practice
It is not a big issue since we don't support these architectures anyway, but I will replace the offending `size_t` instances.

Reply:
https://stackoverflow.com/questions/1464174/size-t-vs-uintptr-t

Reply:
This is another link showing that I should not assume a pointer fits into `size_t`

Reply:
yep, uintptr_t was created exactly for this situation:

http://en.cppreference.com/w/c/types/integer
uintptr_t: unsigned integer type capable of holding a pointer